### DISCUSSION TITLE
machine moving in two axis

### DISCUSSION THREAD

**Discussion by marcootaviomello** [arkypita/LaserGRBL#1487] [Q&A]
I'm modifying a laser machine, which didn't use grbl firmware, used proprietary software along with proprietary company firmware. I recorded the GRBL on it, but when testing with the LASER GRBL when I click on the diagonal movement arrows, that is, it has command in both axes, it moves correctly, but when I click on the conventional up, down, left, right buttons, the same moves diagonally. ME seems to be something from mechanical construction, but can I solve it via GRBL configuration, to send pulse to both engines?

**Comment by arkypita** [comment]
Picture of machine, controller, and mechanical parts please

**Comment by marcootaviomello** [comment]
Analyzing the mechanics, this machine has only 2 motors, to make movements in a straight line I must send a command in both axes, (which is when we click on the arrow to move diagonally on the LASER GRBL), if I send a command only in one of the axes, the machine moves diagonally to the command axis. Any configuration ideas?
![WhatsApp Image 2021-06-03 at 08 56 53](https://user-images.githubusercontent.com/67208525/120641774-69685000-c44a-11eb-9e89-621c11c723f8.jpeg)
![WhatsApp Image 2021-06-03 at 08 56 52 (1)](https://user-images.githubusercontent.com/67208525/120641778-6a997d00-c44a-11eb-9adb-58237fd0eec1.jpeg)
![WhatsApp Image 2021-06-03 at 08 56 52](https://user-images.githubusercontent.com/67208525/120641780-6a997d00-c44a-11eb-957f-3f8288e67e09.jpeg)
![WhatsApp Image 2021-06-03 at 08 56 53 (1)](https://user-images.githubusercontent.com/67208525/120641782-6b321380-c44a-11eb-8bc4-6dfc202c37d7.jpeg)

**Comment by BeatSlayer** [comment]
That's a coreXY type of machine.

**Comment by arkypita** [comment]
Exactly as @BeatSlayer say, it is a CoreXY
You need to flash a GRBL firmware compiled/configured for CoreXY (google it)

**Comment by marcootaviomello** [comment]
Thanks a lot.
